File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        # Find the maximum number of candies any kid has.
        maxCandies = max(candies)
        # Initialize an empty list to store the result (True/False for each kid).
        result = []
        
        # For each kid, check if their candies plus the extraCandies is greater than or equal to the maximum candies found.
        for i in range(len(candies)):            
            result.append(candies[i] + extraCandies >= maxCandies)
            
        # Return the final list of boolean values.
        return result

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        // Initialize variable to store the greatest number of candies among all the kids.
        int maxCandies = 0;  
        // Find the maximum number of candies each kid has.
        for (int candy : candies) {
            maxCandies = Math.max(candy, maxCandies);
        }
        // Create a list to store the result (true/false for each kid).
        List<Boolean> result = new ArrayList<>();
        
        // For each kid, check if their candies plus the extraCandies will be greater than or equal to the maximum candies found.
        for (int candy : candies) {
            result.add(candy + extraCandies >= maxCandies);
        }

        // Return the final list of boolean values.
        return result;
    }
}
